ios/testmanagerd/xcuitestrunner.go

```diff
@@ -285,160 +287,160 @@ var (
 	closedChan = make(chan interface{})
 )
 
-func runXUITestWithBundleIdsXcode12Ctx(ctx context.Context, bundleID string, testRunnerBundleID string, xctestConfigFileName string,
-	device ios.DeviceEntry, conn *dtx.Connection, args []string, env []string,
+func RunXCUIWithBundleIdsCtx(
+	ctx context.Context,
+	bundleID string,
+	testRunnerBundleID string,
+	xctestConfigFileName string,
+	device ios.DeviceEntry,
+	wdaargs []string,
+	wdaenv []string,
 ) error {
-	testSessionId, xctestConfigPath, testConfig, testInfo, err := setupXcuiTest(device, bundleID, testRunnerBundleID, xctestConfigFileName)
+	version, err := ios.GetProductVersion(device)
 	if err != nil {
 		return err
 	}
-	defer conn.Close()
-	ideDaemonProxy := newDtxProxyWithConfig(conn, testConfig)
+	log.Debugf("%v", version)
+	if version.LessThan(ios.IOS14()) {
+		log.Infof("iOS version: %s detected, running with ios11 support", version)
+		return RunXCUIWithBundleIdsXcode11Ctx(ctx, bundleID, testRunnerBundleID, xctestConfigFileName, device, wdaargs, wdaenv)
+	}
 
-	conn2, err := dtx.NewConnection(device, testmanagerdiOS14)
-	if err != nil {
-		return err
+	if version.LessThan(ios.IOS17()) {
+		log.Infof("iOS version: %s detected, running with ios14 support", version)
+		return RunXUITestWithBundleIdsXcode12Ctx(ctx, bundleID, testRunnerBundleID, xctestConfigFileName, device, wdaargs, wdaenv)
 	}
-	defer conn2.Close()
-	log.Debug("connections ready")
-	ideDaemonProxy2 := newDtxProxyWithConfig(conn2, testConfig)
-	ideDaemonProxy2.ideInterface.testConfig = testConfig
-	caps, err := ideDaemonProxy.daemonConnection.initiateControlSessionWithCapabilities(nskeyedarchiver.XCTCapabilities{})
+
+	log.Infof("iOS version: %s detected, running with ios17 support", version)
+	return runXUITestWithBundleIdsXcode15Ctx(ctx, bundleID, testRunnerBundleID, xctestConfigFileName, device, wdaargs, wdaenv)
+}
+
+func CloseXCUITestRunner() error {
+	var signal interface{}
+	go func() { closeChan <- signal }()
+	select {
+	case <-closedChan:
+		return nil
+	case <-time.After(5 * time.Second):
+		return fmt.Errorf("Failed closing, exiting due to timeout")
+	}
+}
+
+func runXUITestWithBundleIdsXcode15Ctx(ctx context.Context,
+	bundleID string,
+	testRunnerBundleID string,
+	xctestConfigFileName string,
+	device ios.DeviceEntry,
+	wdaargs []string,
+	wdaenv []string,
```

Comment:
There are a couple of unused arguments.
Looks like the context is mainly for stopping the execution, even if we can't do that at the moment we could prepare for it already.
And the WDA related things only get merged into the other args, so that should be possible

Reply:
Removed context and wda args. We will need to think about proper interruption for tests but I guess that's for another PR.